release.sh

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+#
+# release.sh - Creates annotated tags for module releases
+#
+# This script is used by maintainers to create annotated tags for module releases.
+# It supports two main modes:
+# 1. Creating a new tag for a module: ./release.sh module-name X.Y.Z
+# 2. Listing all modules with their versions: ./release.sh --list
+#
+# When a tag is pushed, it triggers a GitHub workflow that updates README versions.
+
+set -euo pipefail
+
+# Handle --list option to show all modules and their versions
+if [[ "$#" -eq 1 && "$1" == "--list" ]]; then
+  # Function to extract version from README
+  extract_version() {
+    grep -o 'version *= *"[0-9]\+\.[0-9]\+\.[0-9]\+"' "$1" | head -1 | grep -o '[0-9]\+\.[0-9]\+\.[0-9]\+' || echo "0.0.0"
+  }
+
+  # Display header for module listing
+  echo "Listing all modules and their latest versions:"
+  echo "--------------------------------------------------"
+  printf "%-30s %-15s %s\n" "MODULE" "README VERSION" "LATEST TAG"
+  echo "--------------------------------------------------"
+  
+  # Loop through all directories that look like modules
+  for dir in */; do
+    if [[ -d "$dir" && -f "${dir}README.md" && "$dir" != ".git/" ]]; then
+      module_name="${dir%/}"
+      # Get version from README
+      readme_version=$(extract_version "${dir}README.md")
+      # Get latest tag for this module
+      latest_tag=$(git tag -l "release/${module_name}/v*" | sort -V | tail -n 1)
+      tag_version=$([ -n "$latest_tag" ] && echo "$latest_tag" | sed 's|release/'"${module_name}"'/v||' || echo "none")
+      # Display module info
+      printf "%-30s %-15s %s\n" "$module_name" "$readme_version" "$tag_version"
+    fi
+  done
+  
+  echo "--------------------------------------------------"
+  exit 0
+fi
+
+# Handle --dry-run option
+DRY_RUN=false
+if [[ "$1" == "--dry-run" ]]; then
```

Comment:
```suggestion
if [[ " $* " == " --dry-run " ]]; then
```
This would allow dry run to be placed anywhere, but the shift logic below would have to change too 😅.